mlir/test/Dialect/Affine/loop-fusion-4.mlir

```diff
@@ -226,3 +226,29 @@ func.func @fuse_higher_dim_nest_into_lower_dim_nest() {
   // PRODUCER-CONSUMER:      return
   return
 }
+
```

Comment:
A small test demonstrating the handling of side-effects when fusing loops with fusion-maximal